protos/proto/aptos/transaction/v1/transaction.proto

```diff
@@ -460,10 +461,28 @@ message AnySignature {
     TYPE_UNSPECIFIED = 0;
     TYPE_ED25519 = 1;
     TYPE_SECP256K1_ECDSA = 2;
+    TYPE_WEBAUTHN = 3;
   }
 
   Type type = 1;
-  bytes signature = 2;
+
+  oneof signature {
```

Comment:
Cc @bowenyang007 @larry-aptos Just wanted to share this for context. Not sure if there's a better way to structure this - feel free to lmk if there is - but this proto type on line 469 is tracking the `AnySignature` enum here:

```rust
#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
pub enum AnySignature {
    Ed25519 {
        signature: Ed25519Signature,
    },
    Secp256k1Ecdsa {
        signature: secp256k1_ecdsa::Signature,
    },
    WebAuthn {
        partial_authenticator_assertion_response: PartialAuthenticatorAssertionResponse,
    },
}
```

And then is subsequently used in convert.rs like so

```rust
fn convert_signature(signature: &Signature) -> transaction::AnySignature {
    match signature {
        Signature::Ed25519(s) => transaction::AnySignature {
            r#type: transaction::any_signature::Type::Ed25519 as i32,
            signature: Some(any_signature::Signature::Ed25519(Ed25519 {
                signature: s.0.clone(),
            })),
        },
        Signature::Secp256k1Ecdsa(s) => transaction::AnySignature {
            r#type: transaction::any_signature::Type::Secp256k1Ecdsa as i32,
            signature: Some(any_signature::Signature::Secp256k1Ecdsa(Secp256k1Ecdsa {
                signature: s.0.clone(),
            })),
        },
        Signature::WebAuthn(s) => transaction::AnySignature {
            r#type: transaction::any_signature::Type::Webauthn as i32,
            signature: Some(any_signature::Signature::Webauthn(WebAuthn {
                partial_authenticator_assertion_response: s.0.clone(),
            })),
        },
    }
}
```

There's no clean way in proto afaict to have an enum with different struct fields for each variant so there is an extra `signature` field in between the enum and each variant. 

Also I decided to keep `partial_authenticator_assertion_response` as bytes since I don't imagine people will need to parse it and I don't think it makes a lot of sense to parse `s` (HexEncodedBytes) to get `signature`, `client_data_json`, and `authenticator_data` but let me know if you think it would be necessary.

Reply:
IIRC, oneof may have some upgradability issues in the future? @larry-aptos

Reply:
what's going on here? I think it is okay to just push the bytes here?

Reply:
I can revert it but this may be more future-proof. Before I was using `partial_authenticator_assertion_response` as the field name instead of `signature` but I changed it back so I could remove the changes. If someone in the future decides to add a new enum variant with additional fields this proto will need to be amended

As an example

```rust
pub enum AnySignature {
    Ed25519 {
        signature: Ed25519Signature,
    },
    Secp256k1Ecdsa {
        signature: secp256k1_ecdsa::Signature,
    },
    NewSignature {
        signature: PartialAuthenticatorAssertionResponse,
        new_field: NewFieldType
    },
}
```

would require a modification to the proto files
Cc @davidiw @larry-aptos @bowenyang007

Reply:
This is all new code that you wrote from before right? If os I don't see an issue with changing this part of the proto? Not sure if this is what you're asking

Reply:
David's saying I don't need to change the proto here. I can keep it the way it was before the change.

I'm saying that David is right, but changing it may be more futureproof.